### PR TITLE
feat(query): create new work item dialog

### PR DESCRIPTION
**Pull request by sahil143**
<strong>Note: If there are pending changes to the PR, prefix the PR title with "WIP" and add the label "DO NOT MERGE"</strong>

### Mandatory
 - [ ] What does this PR do? 
- Add a way to create workitem from query page
- [ ] What issue/task does this PR references?
https://openshift.io/openshiftio/Openshift_io/plan/detail/531
- [ ] Are the tests Included?
___
### Optional
- [ ] Is the documentation Included?

- [ ] Are the Release Notes included?

- [ ] @mention(s) to expected reviewer(s) included

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @sahil143 for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


Your plugin configuration is stored in the [file](https://github.com/fabric8-ui/fabric8-planner/blob/3e6e16e7d7c8612fd3f7b5520239fde7d3280ab6/.ike-prow/test-keeper.yaml).

**Comment by divyanshiGupta** [comment]
@sahil143 this PR has lint errors, please resolve them. And is the PR ready for review or is it work-in-progress?

**Comment by divyanshiGupta** [comment]
@sahil143 functional tests are failing.

**Comment by sahil143** [comment]
> @sahil143 functional tests are failing.

Yes @divyanshiGupta I'm working on them

File: src/app/effects/work-item.effects.ts [hunk 1]
```diff
@@ -114,12 +116,23 @@ export class WorkItemEffects {
                     })
                   );
               } else {
-                // for a normal (not a child) work item creation
-                // Add item success notification
+                let currentURL = document.location.pathname;
+                let detailURL = currentURL.indexOf('/plan/query') > -1 ? currentURL.split('/plan/query')[0] : currentURL;
+                const routeURL = detailURL + '/plan/detail/' + wItem.number;
```
Comment (Raunak1203) [inline src/app/effects/work-item.effects.ts]: @sahil143 do we need plan here in '/plan/detail/' ?

**Comment by Raunak1203** [comment]
[test]

File: src/app/components_ngrx/work-item-quick-add/work-item-quick-add.component.html [hunk 2]
```diff
@@ -73,6 +73,64 @@
 </div>
 <!-- Table Inline Quick Add Ends -->
 
+<!--Quick Add for Query Tab-->
+
+<div class="f8-quickadd-query__wrapper" *ngIf="wilistview === 'wi-query-view'">
+  <input class=""
```
Comment (divyanshiGupta) [inline src/app/components_ngrx/work-item-quick-add/work-item-quick-add.component.html]: Why is class attribute empty here? Or if we do not need this you can remove this.

File: src/app/components_ngrx/work-item-quick-add/work-item-quick-add.component.ts [hunk 3]
```diff
@@ -55,6 +56,7 @@ export class WorkItemQuickAddComponent implements OnInit, OnDestroy, AfterViewIn
   linkObject: object;
   addDisabled: Observable<boolean> =
     this.permissionQuery.isAllowedToAdd();
+  workItemTitle= new FormControl('');
```
Comment (divyanshiGupta) [inline src/app/components_ngrx/work-item-quick-add/work-item-quick-add.component.ts]: add space between `workItemTitle` and `=`

**Comment by divyanshiGupta** [comment]
@sahil143 I have added a few review comments here and there apart from that the code looks good to me. And please make sure to remove console logs before merging this to master.

**Comment by sahil143** [comment]
@Raunak1203 
> If a newly created work item is not matching the query, Can we display some notification saying that "Work Item is created but It doesn't match the Query"

Newly created workitem is not shown at all in the list. we only show a notification.

> Create and CreateAndOpen button is disabled after creating 1 workitem

This is fixed in commit https://github.com/fabric8-ui/fabric8-planner/pull/2791/commits/13efa602aba94ec9b38b8420c2122b8c84763cec

> After clicking create button creating a workitem dropdown should close.

No, It should not be because user might want to add another workitem right after a workitem is created

**Pull request merged**